pkgs/by-name/mo/moneydance/package.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, stdenv, fetchurl, makeWrapper, dpkg, openjdk21, openjfx21, }:
+let
+  version = "2023.3_5064";
+  hash = "sha256-t1jBZJ2sL0YDx9915UWgpCzKnpNq1Qz5i7ygGXDI5dk=";
+in stdenv.mkDerivation rec {
+  pname = "moneydance";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://infinitekind.com/stabledl/${version}/moneydance_linux_amd64.deb";
+    inherit hash;
+  };
+
+  jdk = openjdk21.override {
+    enableJavaFX = true;
+  };
+
+  nativeBuildInputs = [ dpkg makeWrapper ];
+  buildInputs = [ jdk openjfx21 ];
+
+  dontConfigure = true;
+  dontUnpack = true;
+  dontBuild = true;
+  dontFixup = true;
+
+  installPhase = let
+    wrapperArgs = lib.concatMapStringsSep " " (flag: "--add-flags ${lib.escapeShellArg flag}") [
+      "-client"
+      "--add-modules"
+      "javafx.swing,javafx.controls,javafx.graphics"
+      "-classpath"
+      "'${placeholder "out"}/libexec/*'" # single-quotes to protect '*' from the shell
+      "Moneydance"
+    ];
+  in ''
+    runHook preInstall
+
+    mkdir -p $out $out/bin
+    dpkg -x $src $out
+    mv $out/opt/Moneydance/lib $out/libexec
+    rm -rf $out/opt
+    makeWrapper ${jdk}/bin/java $out/bin/moneydance ${wrapperArgs}
+
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://infinitekind.com/moneydance";
+    description =
+      "Moneydance is an easy to use and full-featured personal finance app that doesn't compromise your privacy";
+    sourceProvenance = [ lib.sourceTypes.binaryBytecode ];
+    license = lib.licenses.unfree;
+    platforms = lib.platforms.all;
```

Comment:
The package mentions `amd64` (and implicitly linux), does it actually work on all platforms?

Reply:
> Does it actually work on all platforms?

Yes. The package from the company is only platform specific because it ships a Java runtime. This package just strips all that out, and all that's left is JAR files and a driver shell script. Any platform with a Java runtime that supports JavaFX works.

Reply:
Then I would use `openjdk`'s platforms.

Reply:
Oh, that's very clever. Done.